Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -86,7 +118,67 @@ private void ReparseRequested(object sender, EventArgs e)
                 token = _currentCancellationTokenSource.Token;
             }
 
+            BeginParse(sender,token);
+        }
+
+        public void SuspendRequested(object sender, RubberduckStatusSuspendParserEventArgs e)
+        {
+            object parseRequestor = null;
 
+            try
+            {
+                if (_parsingSuspendLock.IsReadLockHeld)
+                {
+                    Logger.Warn("A suspension action was attempted while a read lock was held.");
+                    e.Declined = true;
+                    return;
+                }
+                _parsingSuspendLock.EnterWriteLock();
+                Interlocked.Add(ref _suspensionIteration, 1);
+                var originalStatus = State.Status;
+                Cancel();
```

Comment:
This is potentially problematic. If a parse has already passed the check for the suspension but not entered the read lock, this parse will get canceled here. If no other parse gets requested during the suspension, no corresponding parse will happen.

Reply:
I see that the scenario described is not relevant anymore since the main read lock has been removed. However, that leads to the scenario that a parse is running or waiting for the cancellation of another parse at the `Monitor.Enter`. In this case, no  reparse would happen unless another parse request gets issued from somewhere.